packages/webpack-plugin/src/plugin-utils.ts

```diff
@@ -0,0 +1,368 @@
+        moduleGraph.getIncomingConnections(m),
+        ({ resolvedOriginModule, dependency }) =>
+            dependency instanceof UnusedDependency ? undefined : resolvedOriginModule
+    );
+
+    // TODO: check if this optimization is good.
+    // const inChunks = chunkGraph.getNumberOfModuleChunks(m)
+    // if(inChunks === 0) {
+    //     debugger
+    // }
+
+    let isInUse = false;
+    for (const cm of inConnections) {
+        if (cm.buildMeta.sideEffectFree) {
+            const info = moduleGraph.getExportsInfo(cm);
+            const usedExports = (info.getUsedExports as any)(/*if passed undefined it finds usages in all chunks*/);
+            if (usedExports === false) {
+                continue;
+            } else if (usedExports === true || usedExports === null) {
+                /** noop */
+            } else if (usedExports.size === 0) {
+                continue;
+            }
+        }
+        const chunksCount = chunkGraph.getNumberOfModuleChunks(cm);
+
+        if (chunksCount > 0) {
+            isInUse = true;
+            break;
+        }
+    }
+    return isInUse;
+}
+
+export function getFileName(filename: string, data: Record<string, string>) {
+    return filename.replace(/\[(.*?)]/g, (fullMatch, inner) => {
+        const [type, len] = inner.split(':');
+        const value = data[type];
+        if (value) {
+            const length = Number(len);
+            return !isNaN(length) ? value.slice(0, length) : value;
+        } else {
+            return fullMatch;
+        }
+    });
+}
+
+export function getSortedModules(stylableModules: Set<NormalModule>) {
+    return Array.from(stylableModules)
+        .sort((m1, m2) => {
+            if (m1.resource > m2.resource) {
+                return 1;
+            } else if (m1.resource < m2.resource) {
+                return -1;
+            } else {
+                return 0;
+            }
+        })
+        .sort((m1, m2) => getStylableBuildMeta(m2).depth - getStylableBuildMeta(m1).depth);
+}
+
+export function reportNamespaceCollision(
+    namespaceToFileMapping: Map<string, Set<string>>,
+    errors: Error[]
+) {
+    for (const [namespace, resources] of namespaceToFileMapping) {
+        if (resources.size > 1) {
+            errors.push(
+                new Error(
+                    `Duplicate namespace ${JSON.stringify(
+                        namespace
+                    )} found in multiple different resources:\n${Array.from(resources)
+                        .map((resource) => {
+                            return resource;
+                        })
+                        .join(
+                            '\n'
+                        )}\nThis issue indicates multiple versions of the same library in the compilation, or different paths importing the same stylesheet like: "esm" or "cjs".`
+                )
+            );
+        }
+    }
+}
+
+type OptimizationMapping = {
+    usageMapping: Record<string, boolean>;
```

Comment:
any reason for the first two to be records while the third is a Map?

Reply:
only one of the needs to be enumerable